### PR TITLE
PT-4026: conflict-note card (Accept/Reject with dynamic Result preview)

### PR DESCRIPTION
**Pull request by rolfheij-sil**
## What & why

Part of epic **PT-4024** (Donna resolves merge conflicts), **stacked on #2482** (the PT-4028 read-contract — this PR's base is `pt-4028-conflict-note-fields`). It adds the presentational conflict-resolution card so a translator can see what is rejected, what is accepted, and how the verse will read after their choice (NN3), and accept or reject it (NN4).

`ConflictNoteCard` — a body fragment in the shared `comment-list` lib — renders:
- a `conflictType`-keyed description,
- an Accept/Reject selector,
- **Rejected** & **Accepted** diff regions (PT9 `<u>`/`<s>` markup, colored via theme tokens),
- a read-only **Result preview that tracks the selection** — `resultText` on accept, `rejectedResultText` on reject.

## PT9-anchored behavior

Matches PT9's `AppendConflictNoteDetails`: the structured card renders when `conflictType === 'verseText' && !!resultText`; the **Accepted region is shown only when `acceptedText` is present**. So **no-common-ancestor conflicts** — two translators independently drafting a previously-absent verse, which have no accepted-side diff — still get the full resolution UI, just without the Accepted panel (the selector + dynamic Result preview still work). Genuinely non-`verseText` notes fall back to the existing `contents` render.

The dynamic Result preview relies on `rejectedResultText` (the loser's plain USFM via PT9's public `GetDiffVerseUsfm`), which was added to the read-contract in the base PR **#2482** (along with a no-ancestor converter test and softened `acceptedText` docs).

## Scope

- **Presentation only** — no PAPI, no live data. The card emits the choice via `onResolutionChange`; `canAcceptReject` gates the selector.
- `verseText` conflicts only (v1).
- Integration (rendering conflict threads with this card, wiring resolve to PT-4029, contributing localized string values) is **PT-4030**. `ConflictNoteCard` + `CONFLICT_NOTE_STRING_KEYS` are exported from `platform-bible-react` for that ticket.

## Testing

- Vitest: 3-region render, dynamic Result toggle (including a **real Radix `Select` interaction** firing `onResolutionChange`), the no-ancestor case (Accepted omitted, selector + Result still work), non-`verseText` fallback, disabled selector, and read-only Result.
- Storybook (`Advanced/ConflictNoteCard`): Default, NoAncestor, RestrictedPermissions. Sample field values are captured **verbatim** from the PT-4028 C# converter — not hand-written.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

- - -
This change is [Reviewable](https://reviewable.io/reviews/paranext/paranext-core/2484)

---

## Review summary (`/code-review max` + `/review-paratext`, 2026-07-03)

Reviewed with a max-effort multi-agent `/code-review` (32 agents → 15 distinct defects) plus the four
`/review-paratext` analyzers. All Critical and Important findings were **fixed in-review** (commit
`9f0c52a`). Branch is green: typecheck, lint, `format:base:check`, unit 230/230, Storybook 3/3, C#
23/23. Full findings (incl. dismissed/deferred) are in the GitHub review on this PR.

> ⚠️ **Supersedes "PT9-anchored behavior" above**: the structured card now renders when
> `conflictType === 'verseText'` **alone** (the `&& !!resultText` gate was removed). A two-agent spike
> against PT9 source confirmed PT9 gates only on `isFirstComment && Contents?.FirstChild != null` and
> never on result-emptiness; that blank-`resultText` state is not even producible from a real merge
> (a deleted verse creates no note; an emptied verse keeps its `\v N` marker). The card now degrades
> gracefully per-region instead of hiding the whole resolution UI.

**Key changes made in review:**

- **Robustness / PT9 parity**: gate on `conflictType` alone; gate the Rejected region on
  `!!rejectedText` (like Accepted); neutral empty-state in the Result preview when the chosen side has
  no result USFM.
- **Ship-blockers**: regenerated the tracked `platform-bible-react` `dist/` (new exports were missing);
  fixed the prettier failure that would have red-CI'd the branch.
- **Dark-mode contrast**: new per-scheme `--diff-deleted` theme token (deletion text was
  background-grade `--destructive`, failing AA in Platform dark).
- **Localization**: frozen `as const` key tuple + derived type + optional `localizedStrings` prop
  (guide-conforming); keys renamed to camelCase `%conflictNote_*%` before PT-4030 locks values in.
- **A11y / quality**: `aria-live` on the Result preview; shared prose classes with `CommentItem`;
  controlled-mode state-leak guard; `min-w` selector; renamed the both-sides C# fixture/test/sample;
  added edge-case + regression tests (11 unit tests).

**Dismissed**: whitespace-only-diff invisibility (degenerate conflict); missing `pr-twp` (family
consistency). **Deferred to PT-4030**: reset uncontrolled state on `comment` change (key the card by
comment id) + contributing the 9 `%conflictNote_*%` string values. **Follow-up**: shared jsdom test
setup (4th copy).

### CODE REVIEW

**Review by rolfheij-sil — COMMENTED** [review]
## Automated code review — `/code-review max` + `/review-paratext`

Reviewed 2026-07-03 with a max-effort multi-agent `/code-review` (32 agents, 36 verified findings → 15 distinct) and the four `/review-paratext` analyzers (api / style / compliance / ux). Where I1/I2/I3 hinged on PT9 behavior, a two-agent spike established ground truth against PT9 source (`ParatextBase`/`ParatextData`) and the PT10 converter before any behavior change.

**Everything found is listed below — including items dismissed or deferred — with how each was handled.** All Critical and Important findings were fixed in-review (commit `9f0c52a`). Branch is green: typecheck · eslint · `format:base:check` · Vitest unit 230/230 · Storybook (chromium) 3/3 · C# `PlatformCommentConverterTests` 23/23. The tracked `dist/` and `themes.data.json` were regenerated/synced.

### Critical

| # | Finding | Handling |
|---|---------|----------|
| C1 | **Stale tracked `dist/`** — new exports were added to `src/index.ts` but the git-tracked build output consumers resolve was not rebuilt, so PT-4030's `import { ConflictNoteCard }` would fail TS2305 / crash at runtime; no CI step catches it. | **Fixed** — regenerated `platform-bible-react` dist; exports now present in `dist/index.d.ts`/`index.js`. |

### Important

| # | Finding | Handling |
|---|---------|----------|
| I1 | **`!!resultText` gate hid the entire resolution UI** when the merged result verse is blank, diverging from PT9 (`AppendConflictNoteDetails` gates only on `isFirstComment && Contents?.FirstChild != null`, never on result-emptiness). | **Fixed** — gate on `conflictType === 'verseText'` alone. *Spike finding:* the trigger state is **not producible from real merge data** (a deleted verse creates no note — `BookFileMerger.cs:441-445`; an emptied verse keeps its `\v N` marker so `Comment.Verse` is non-blank). So this was a defensive-but-wrong gate, not a live stranding bug — but the fix restores PT9 parity and hardens the card. |
| I2 | **Rejected region heading rendered unconditionally** → orphan "Rejected" heading over an empty body when `rejectedText` is absent. | **Fixed** — gated on `!!rejectedText`, symmetric with the Accepted region (matches PT9, which prints the reject label only when the rejected side is non-null). |
| I3 | **Blank `<p>` in the Result preview** when Reject is selected and `rejectedResultText` is absent. | **Fixed** — neutral empty-state string ("The verse will be empty.") instead of an empty paragraph. |
| I4 | **Deletion text below AA contrast in Platform dark mode** — `<s>` used background-grade `--destructive` (~2–2.5:1), while insertions used the text-grade `--success-foreground`. | **Fixed** — added a per-scheme text-grade `--diff-deleted` token (red-600 light / red-400 dark) in `index.css`; `themes.data.json` auto-synced; `<s>` now uses it. |
| I5 | **CI prettier/format gate would fail** on the component + sample data (root `--check` misses it via `.prettierignorerun`; the lib workspace check exits 1). | **Fixed** — reformatted; whole-lib `format:base:check` passes. |
| I6a | **Localization contract deviated from the Localization-Guide** — mutable `LocalizeKey[]` + required `localizedStrings` prop (guide/newer components use a frozen tuple + derived type + optional prop). | **Fixed** — `Object.freeze([...] as const)`, derived `ConflictNoteCardLocalizedStrings`, optional prop. |
| I6b | **Snake_case key prefix** `%conflict_note_*%` (guide prefers camelCase; keys are immutable once shipped). | **Fixed** — renamed to `%conflictNote_*%` before PT-4030 contributes values. |
| I6c | **`?? 'English'` fallbacks are dead through the documented hook path** (`useLocalizedStrings` returns missing keys as the key itself). | **Addressed** by the contract change; the fallbacks remain a defensive default. Contributing string values is PT-4030's job. |
| I7 | **Fallback path skipped `trimDiffSpanWhitespace`** while styling with diff classes → potential dangling strikethrough. | **Obsoleted** by the I1 fix — the fallback now renders only non-verseText conflicts, which carry message-only contents with no `<s>`/`<u>` markup. |
| I8 | **Result preview not announced to assistive tech** — no `aria-live`; screen-reader users don't hear the Accept↔Reject swap. | **Fixed** — `aria-live="polite"` region + `aria-labelledby` to the Result label, keyed with `useId` for list uniqueness. |
| I9 | **`DIFF_HTML_CLASSES` copy-pasted CommentItem's prose/blockquote stack** — silent drift risk. | **Fixed** — extracted `COMMENT_BODY_PROSE_CLASSES`; both consume it. |
| I10 | **Confusing C# fixture naming** — new `CreateVerseTextConflictCommentWithReplacement` vs base's `…Replacement`; TS sample named after the wrong fixture. | **Fixed** — renamed the fixture, its test, and the sample to `…ReplacementBothSides`. |
| I11 | **Missing edge-case tests** for the new gating/absent-field logic. | **Fixed** — added tests for verseText-without-`resultText`, absent-`rejectedText`, absent-`rejectedResultText`-on-reject, aria-live, and the controlled-mode state-leak (11 unit tests total). |

### Minor

| # | Finding | Handling |
|---|---------|----------|
| M1 | **Whitespace-only diff invisible** — `trimDiffSpanWhitespace` collapses `<s> </s>` to an empty element. | **Dismissed** — a whitespace-only conflict is degenerate; the trim is deliberate for the common case. |
| M2 | **Controlled-mode internal-state leak** — `handleChange` wrote `internalResolution` even when controlled, so a declined choice could resurface if the parent later stopped controlling. | **Fixed** — write internal state only when uncontrolled; test added. |
| M3 | **Uncontrolled state not reset on `comment` change** — a reused instance leaks the prior conflict's choice. | **Deferred to PT-4030** — the list integration should key the card by comment id (not a presentational-card concern). |
| M4 | **`w-32` clips long localized selector values** (no tooltip fallback). | **Fixed** — `min-w-32` keeps the stable width without clipping. |
| M5 | **Missing `pr-twp` on the root element.** | **Dismissed** — the whole comment-list family omits it and the card only renders inside `CommentList`; matching the family is correct. |
| M6 | **4th verbatim copy of the jsdom `NoopResizeObserver`/pointer stubs** across test files. | **Deferred to a follow-up** — a shared `setupFiles` touches 4 test files + vitest config, out of scope here. |
| M7 | **"captured verbatim / do not hand-edit" JSDoc** contradicted by hand-renamed `id`/`thread`. | **Fixed** — JSDoc now scopes the verbatim claim to the HTML/USFM fields and notes the readability rename. |

The workflow also surfaced ~6 additional lowest-severity cleanup candidates that fell under its 15-decision cap; the ones that mattered are captured above (I9, M6).

### Positive observations

PT9 anchoring verified against real PT9 source (region order, accepted-region gating, default `'accept'`, permission gating); purely additive public API with all referenced types exported; DOMPurify on every HTML injection; RTL-safe logical properties; redundant color encoding (semibold/strikethrough) for color-blind users; strong verified-green tests including a real Radix `Select` interaction; sample data cross-anchored to a matching C# fixture + converter test.

### Suggested reviewer focus

- The **I1 gate change** (drop `&& !!resultText`) — confirm you agree it's the right PT9-parity call; the spike concluded the blank-`resultText` state is unproducible from real merges.
- The new **`--diff-deleted` theme token** — verify the red-600/red-400 values and that nothing else expected `<s>` to use `--destructive`.
- The **localization key rename** (`%conflictNote_*%`, camelCase) — effectively immutable once PT-4030 ships values.

*Automated review by Claude Fable 5 (analysis) and Claude Opus 4.8 (spike + fixes).*

File: lib/platform-bible-react/src/components/advanced/comment-list/conflict-note-card.component.tsx [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+        {/* aria-live so screen readers announce the Result changing when the user switches
+            Accept/Reject; on a stable wrapper (not the swapped <p>) so the update is detected. */}
+        <div aria-live="polite" aria-labelledby={resultLabelId}>
+          {resultText ? (
+            <p className="tw:whitespace-pre-wrap tw:text-foreground">{resultText}</p>
+          ) : (
+            // The chosen side has no result USFM (e.g. reject deletes the verse). Show a neutral
+            // empty-state rather than a blank <p>, which reads as a rendering bug.
+            <p className="tw:italic tw:text-muted-foreground">
+              {localizedStrings?.['%conflictNote_resultEmpty%'] ?? 'The verse will be empty.'}
```
Comment (lyonsil) [inline lib/platform-bible-react/src/components/advanced/comment-list/conflict-note-card.component.tsx]: The empty-state hardcodes **"The verse will be empty."** for any falsy result value, but this PR's own TSDoc for `rejectedResultText` (`comments.types.ts`) documents two distinct absence causes: the reject outcome decodes to an empty verse, *or* "the note carries no decodable diff." Only the first justifies asserting emptiness. On Reject with `rejectedResultText` absent for the no-decodable-diff reason, the card states an emptiness that isn't known to hold. Consider showing the definitive "will be empty" message only for the empty-verse case and a neutral "no preview available" otherwise.

File: lib/platform-bible-react/src/components/advanced/comment-list/conflict-note-card.component.test.tsx [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+  }
+  if (!Element.prototype.hasPointerCapture) {
+    Element.prototype.hasPointerCapture = () => false;
+  }
+  if (!Element.prototype.scrollIntoView) {
+    Element.prototype.scrollIntoView = () => {};
+  }
+});
+
+const localizedStrings = {
```
Comment (lyonsil) [inline lib/platform-bible-react/src/components/advanced/comment-list/conflict-note-card.component.test.tsx:49]: Every value in this `localizedStrings` object (and the identical one in `conflict-note-card.stories.tsx`) is byte-for-byte equal to the component's hardcoded English fallbacks. Because provided value == fallback for every key, the rendered DOM is identical whether or not the `localizedStrings?.[...] ?? '...'` lookups are actually consumed — so no test or story here can catch a dropped or mistyped key (it would silently fall through to the correct fallback and stay green). Giving at least one key a sentinel value that differs from the fallback (e.g. `'%conflictNote_accept%': 'ACCEPT_XYZ'`) and asserting on it would make the localization wiring falsifiable.

File: lib/platform-bible-react/src/components/advanced/comment-list/conflict-note-card.component.tsx [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+            <SelectItem value="reject">
+              {localizedStrings?.['%conflictNote_reject%'] ?? 'Reject'}
+            </SelectItem>
+          </SelectContent>
+        </Select>
+      </div>
+
+      {/* Accepted region first, then Rejected — matches PT9's AppendConflictNoteDetails order
+          (winning/current text leads, and Accept is the default resolution). */}
+      {!!comment.acceptedText && (
```
Comment (lyonsil) [inline lib/platform-bible-react/src/components/advanced/comment-list/conflict-note-card.component.tsx:134]: The Accepted region here and the Rejected region just below are near-identical — they differ only in the gate field, the label, and which sanitized HTML is injected — and the `<div className={DIFF_HTML_CLASSES} dangerouslySetInnerHTML>` + `eslint-disable react/no-danger` block is repeated three times (here, Rejected, and the fallback). Extracting a small `DiffRegion({ label, rawHtml })` (sanitizing/trimming internally) would collapse the two regions into one and remove two of the three `react/no-danger` disables. Worth doing specifically because it's a `dangerouslySetInnerHTML` pattern, where a divergent edit to one copy is silent.

File: lib/platform-bible-react/src/components/advanced/comment-list/conflict-note-card.component.tsx [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+
+  const sanitizedRejected = useMemo(
+    () => trimDiffSpanWhitespace(sanitizeHtml(comment.rejectedText ?? '')),
+    [comment.rejectedText],
+  );
+  const sanitizedAccepted = useMemo(
+    () => trimDiffSpanWhitespace(sanitizeHtml(comment.acceptedText ?? '')),
+    [comment.acceptedText],
+  );
+  const sanitizedContents = useMemo(() => sanitizeHtml(comment.contents), [comment.contents]);
```
Comment (lyonsil) [inline lib/platform-bible-react/src/components/advanced/comment-list/conflict-note-card.component.tsx]: `sanitizeHtml` is a full DOMPurify parse, and it runs here on `comment.contents` for every card — but only the non-verseText fallback renders `sanitizedContents`, so on a verseText card the parse is discarded. Symmetrically, `sanitizedRejected` / `sanitizedAccepted` are computed on non-verseText cards that never render them. Since `isVerseTextConflict` is a cheap string compare, hoisting it above these memos and gating each memo body on it would avoid the wasted parse while keeping the hooks unconditional. Minor (one discarded parse per card mount) — noting for completeness.

File: lib/platform-bible-react/src/components/advanced/comment-list/conflict-note-card.types.ts [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+  '%conflictNote_accept%',
+  '%conflictNote_reject%',
+  '%conflictNote_rejectedLabel%',
+  '%conflictNote_acceptedLabel%',
+  '%conflictNote_resultLabel%',
+  '%conflictNote_resultEmpty%',
+] as const);
+
+/** Type definition for the localized strings used in the ConflictNoteCard component */
+export type ConflictNoteCardLocalizedStrings = {
```
Comment (lyonsil) [inline lib/platform-bible-react/src/components/advanced/comment-list/conflict-note-card.types.ts:27]: `ConflictNoteCardLocalizedStrings` is defined and module-exported here, but the `localizedStrings` prop is typed as the loose `LanguageStrings` (line 39), and this type isn't re-exported from `index.ts` — so nothing uses it and package consumers can't import it. An extension author wiring up the card via `CONFLICT_NOTE_STRING_KEYS` gets no key-level type safety (a typo'd key compiles clean and silently renders the fallback). Either apply it to the prop and re-export it (as ~13 other advanced components do with their `*LocalizedStrings` types) or drop it. Note: sibling `CommentListLocalizedStrings` has the same gap, so there's local precedent either way.

File: lib/platform-bible-react/src/components/advanced/comment-list/conflict-note-card.component.tsx [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+// Shares COMMENT_BODY_PROSE_CLASSES with CommentItem so conflict notes render note-body HTML the same
+// way, then layers theme-token diff coloring on top: <u> = inserted → green + semibold; <s> = deleted →
+// red + strikethrough. Green/red text-color treatment (theme tokens only, no hardcoded colors) — NOT a
+// copy of any single PT9 diff style: PT9's default (BlueGray) and RedGreen both use background
+// highlights, and RedGreen has no strikethrough. The <u>/<s> markup comes from PT9's decode; only the
+// coloring is ours. Deletions use --diff-deleted (a per-scheme text-grade red), not --destructive,
+// which is background-grade in the Platform dark theme and fails text contrast there.
+const DIFF_HTML_CLASSES = cn(
+  COMMENT_BODY_PROSE_CLASSES,
+  'tw:[&_u]:font-semibold tw:[&_u]:text-success-foreground tw:[&_u]:no-underline',
```
Comment (lyonsil) [inline lib/platform-bible-react/src/components/advanced/comment-list/conflict-note-card.component.tsx:26]: Minor/optional consistency nit: `<u>` insertions reuse the general `--success-foreground` token (also the toolbar success check), while `<s>` deletions get the purpose-built `--diff-deleted`. Retoning the shared success token would silently shift diff-insertion color here, and tuning only diff-insertion green would force editing a token shared with form-validation/toolbar. A symmetric `--diff-inserted` would keep the pair at the same altitude. The asymmetry is defensible given the rationale in the comment above (no text-grade red existed and `--destructive` is background-grade), so this is really just a token-altitude nit.

**Review by lyonsil — COMMENTED** [review]
Follow-up review pass (automated multi-agent review, human-curated). A few non-blocking notes inline — one correctness edge case around the Result empty-state, the rest test-coverage / cleanup / consistency. None block merge for this PR's presentation-only scope.

**Review by lyonsil — COMMENTED** [review]
Claude found a few things, and I didn't see anything else in my scan of the changes. I ignored some of the Claude findings and only posted the ones that seemed potentially valid. None of them are really serious, but the ones I posted seem considering at least.

> @lyonsil reviewed 22 files and all commit messages, and made 1 comment.
> *[Reviewable](https://reviewable.io/reviews/paranext/paranext-core/2484#-:-OwxepOzBnylkCQXBHch:b-tpzvb2)* status: all files reviewed, 6 unresolved discussions (waiting on rolfheij-sil).

File: lib/platform-bible-react/src/components/advanced/comment-list/conflict-note-card.component.tsx (hunk 1 above)
Comment (rolfheij-sil) [reply lib/platform-bible-react/src/components/advanced/comment-list/conflict-note-card.component.tsx]: > 🤖 _Drafted by Claude (Rolf's AI assistant) and posted on his behalf._

Good catch — fixed in `42168914`. The message asserted an emptiness we can't actually confirm: `rejectedResultText`/`resultText` are absent for either the empty-verse *or* the no-decodable-diff cause, and the card sees a single `undefined` for both. Since there's no signal to branch on, I made the message unconditionally neutral — **"No result preview available."** Renamed the key `%conflictNote_resultEmpty%` → `%conflictNote_resultUnavailable%` to match, since values aren't shipped yet (PT-4030 contributes them).

File: lib/platform-bible-react/src/components/advanced/comment-list/conflict-note-card.component.test.tsx (hunk 2 above)
Comment (rolfheij-sil) [reply lib/platform-bible-react/src/components/advanced/comment-list/conflict-note-card.component.test.tsx:49]: > 🤖 _Drafted by Claude (Rolf's AI assistant) and posted on his behalf._

Fixed in `42168914`. Gave `%conflictNote_description_verseText%` a sentinel value that differs from the fallback, plus an assertion — so a dropped or mistyped key now fails instead of silently rendering the identical fallback. Left the rest matching their fallbacks since other tests exercise the omitted-prop path.

File: lib/platform-bible-react/src/components/advanced/comment-list/conflict-note-card.component.tsx (hunk 3 above)
Comment (rolfheij-sil) [reply lib/platform-bible-react/src/components/advanced/comment-list/conflict-note-card.component.tsx:134]: > 🤖 _Drafted by Claude (Rolf's AI assistant) and posted on his behalf._

Done in `42168914`. Extracted `DiffRegion({ label?, rawHtml })`; the three `dangerouslySetInnerHTML` + `eslint-disable` blocks collapse to one audited sink. Sanitize/trim moved inside, so the divergent-edit risk is gone.

File: lib/platform-bible-react/src/components/advanced/comment-list/conflict-note-card.component.tsx (hunk 4 above)
Comment (rolfheij-sil) [reply lib/platform-bible-react/src/components/advanced/comment-list/conflict-note-card.component.tsx]: > 🤖 _Drafted by Claude (Rolf's AI assistant) and posted on his behalf._

Resolved as a side effect of the DiffRegion extraction (`42168914`): sanitize now lives inside `DiffRegion`, which only mounts for a region that actually renders — so there's no discarded parse on a verseText card or the fallback. The three parent-level memos are gone.

File: lib/platform-bible-react/src/components/advanced/comment-list/conflict-note-card.types.ts (hunk 5 above)
Comment (rolfheij-sil) [reply lib/platform-bible-react/src/components/advanced/comment-list/conflict-note-card.types.ts:27]: > 🤖 _Drafted by Claude (Rolf's AI assistant) and posted on his behalf._

Fixed in `42168914` — went with apply + re-export: the `localizedStrings` prop is now typed `ConflictNoteCardLocalizedStrings` and the type is re-exported from `index.ts`, matching the sibling components, so PT-4030 gets key-level safety. Left `CommentListLocalizedStrings`'s identical gap alone as out of scope here — happy to fix it in a follow-up for consistency.

File: lib/platform-bible-react/src/components/advanced/comment-list/conflict-note-card.component.tsx (hunk 6 above)
Comment (rolfheij-sil) [reply lib/platform-bible-react/src/components/advanced/comment-list/conflict-note-card.component.tsx:26]: > 🤖 _Drafted by Claude (Rolf's AI assistant) and posted on his behalf._

Leaving this one as-is for now. As you noted the asymmetry is defensible: there was no text-grade red, so `--diff-deleted` was purpose-built, whereas `--success-foreground` is a legitimate text-grade green insertions can share. A dedicated `--diff-inserted` only pays off once we need to retone insertions independently of the success token — I'd rather add it then than carry a token (plus the `themes.data.json` churn) purely for symmetry. Easy to revisit if you'd prefer the pair at the same altitude.

File: lib/platform-bible-react/src/components/advanced/comment-list/conflict-note-card.types.ts (hunk 5 above)
Comment (rolfheij-sil) [reply lib/platform-bible-react/src/components/advanced/comment-list/conflict-note-card.types.ts:27]: > 🤖 _Drafted by Claude (Rolf's AI assistant) and posted on his behalf._

Follow-up: fixed the sibling gap in `c1e6392`. Made `COMMENT_LIST_STRING_KEYS` a `const` tuple so `CommentListLocalizedStrings` actually carries the literal key union (it was collapsing to `LanguageStrings` because the keys were typed `LocalizeKey[]`), and re-exported the type. Kept the three `localizedStrings` props as `LanguageStrings` deliberately — the CommentList family indexes keys without fallbacks, so a required mapped type breaks internal access while an optional one rejects the index-signature `LanguageStrings` from `useLocalizedStrings` (TS2740). Same tradeoff `comment-editor` makes; both key consumers go through `Array.from`, so the readonly tuple is transparent.

File: lib/platform-bible-react/src/components/advanced/comment-list/conflict-note-card.component.tsx (hunk 6 above)
Comment (lyonsil) [reply lib/platform-bible-react/src/components/advanced/comment-list/conflict-note-card.component.tsx:26]: No worries - I thought this was reasonable to consider but not required.

**Review by lyonsil — APPROVED** [review]
:lgtm:

> @lyonsil reviewed 13 files and all commit messages, made 1 comment, and resolved 6 discussions.
> *[Reviewable](https://reviewable.io/reviews/paranext/paranext-core/2484#-:-Owy_oiuEblR0HtwYCSW:bnfp4nl)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved.
